### PR TITLE
Fixes #14694: Make ncf-setup able to run the ncf tests from a ncf pull request

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/14694

### CODE REVIEW

**Comment by Fdall** [comment]
Commit modified

File: scripts/rudder-setup/ncf-setup-src.sh [hunk 1]
```diff
@@ -35,18 +35,38 @@ usage() {
 
 setlocal || re_exec "$@"
 
+# Parsing the options before anything, and removing them from $@
+SERVERSPEC=0
+for opt in "$@"
+do
+    shift
+    case "$opt" in
+      --server-spec)
```
Comment (peckpeck) [inline scripts/rudder-setup/ncf-setup-src.sh]: This option is not documented

File: scripts/rudder-setup/ncf-setup-src.sh [hunk 2]
```diff
@@ -35,18 +35,38 @@ usage() {
...
+        SERVERSPEC=1
+        ;;
+      *)
+        set -- "$@" "$opt"
```
Comment (peckpeck) [inline scripts/rudder-setup/ncf-setup-src.sh]: you must break here or adding --server-spec as the last option will break something

File: scripts/rudder-setup/ncf-setup-src.sh [hunk 3]
```diff
@@ -35,18 +35,38 @@ usage() {
...
+        ;;
+    esac
+done
+
 COMMAND="$1"
 NCF_VERSION="$2"
 CFENGINE_VERSION="$3"
 
+[ -z "${CFENGINE_VERSION}" ] && usage
```
Comment (peckpeck) [inline scripts/rudder-setup/ncf-setup-src.sh]: This one

File: scripts/rudder-setup/ncf-setup-src.sh [hunk 4]
```diff
@@ -35,18 +35,38 @@ usage() {
...
+
 if [ -z "$3" ]
```
Comment (peckpeck) [inline scripts/rudder-setup/ncf-setup-src.sh]: conflicts with this one

Comment (peckpeck) [reply scripts/rudder-setup/ncf-setup-src.sh]: This one should probably be $4

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

**Pull request merged**